src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -96,15 +96,46 @@ void DetachableJoint::Configure(const Entity &_entity,
   }
 
   // Setup detach topic
-  std::vector<std::string> topics;
-  if (_sdf->HasElement("topic"))
+  std::vector<std::string> detachTopics;
+  if (_sdf->HasElement("detach_topic"))
   {
-    topics.push_back(_sdf->Get<std::string>("topic"));
+    detachTopics.push_back(_sdf->Get<std::string>("detach_topic"));
   }
-  topics.push_back("/model/" + this->model.Name(_ecm) +
+  detachTopics.push_back("/model/" + this->model.Name(_ecm) +
       "/detachable_joint/detach");
-  this->topic = validTopic(topics);
+  this->detachTopic = validTopic(detachTopics);
 
+  // Setup subscriber for detach topic
+  this->node.Subscribe(
+      this->detachTopic, &DetachableJoint::OnDetachRequest, this);
+
+  ignmsg << "DetachableJoint subscribing to messages on "
+         << "[" << this->detachTopic << "]" << std::endl;
+
+  // Setup attach topic
+  std::vector<std::string> attachTopics;
+  if (_sdf->HasElement("attach_topic"))
+  {
+    attachTopics.push_back(_sdf->Get<std::string>("attach_topic"));
+  }
+  attachTopics.push_back("/model/" + this->model.Name(_ecm) +
+      "/attachable_joint/attach");
+  this->attachTopic = validTopic(attachTopics);
```

Comment:
this might return an empty string, do you mind to check it ?

Reply:
you mean `this->attachTopic`? If it has `attach_topic` string in the sdf file, it'll use that to create a topic. If not, it'll use the `/modle/ this->model.Name(_ecm) + /attachable_joint/attach` so it shouldn't be an empty string. I ran it with a dbg statement and you can see that it isn't an empty string

```
  std::vector<std::string> attachTopics;
  if (_sdf->HasElement("attach_topic"))
  {
    attachTopics.push_back(_sdf->Get<std::string>("attach_topic"));
  }
  attachTopics.push_back("/model/" + this->model.Name(_ecm) +
      "/attachable_joint/attach");
  this->attachTopic = validTopic(attachTopics);
  ```
  
  ```
 [Dbg] [DetachableJoint.cc:107] Detach topic is: /B1/detach
[Msg] DetachableJoint subscribing to messages on [/B1/detach]
[Dbg] [DetachableJoint.cc:125] Attach topic is: /B1/attach
```